rs/ic_os/config/src/types.rs

```diff
@@ -120,3 +114,37 @@ pub struct Logging {
     /// Space-separated list of tags to apply to exported log records.
     pub elasticsearch_tags: Option<String>,
 }
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct NetworkSettings {
+    pub ipv6_config: Ipv6Config,
+    pub ipv4_config: Option<Ipv4Config>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct Ipv4Config {
+    pub address: Ipv4Addr,
+    pub gateway: Ipv4Addr,
+    pub prefix_length: u8,
+    pub domain: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum Ipv6Config {
+    Deterministic(DeterministicIpv6Config),
+    Fixed(FixedIpv6Config),
+    RouterAdvertisement,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
```

Comment:
nit: I _think_ these types have direct equivalence as well, could you add an `Eq` derive too?

Reply:
Added!